machinist/mnode/mnode.go

```diff
@@ -67,3 +76,91 @@ func (n *Node) BeginPolling() error {
 		}
 	}
 }
+
+// Todo(adam): perform rollbacks if enroll fails
```

Comment:
(not a requirement for this PR)

A simple way to implement rollback could be:
1) Array of functions to undo, simple `defer` invoking each function in order.
2) Each file modified, you have a function saving the original in a /tmp/ directory. Eg, a function like "SaveForRollback(path) which saves a file in /tmp/ directory, returns callback to invoke in defer that puts back the original file in place".
3) Each file created, you have a function deleting it.

(should not be too hard to implement)

Reply:
Yeah not difficult at all, but it would significantly increase the LOC of this PR. So I'll add it later.